internal/resources/cluster/data_source_cluster.go

```diff
@@ -25,12 +25,14 @@ import (
 
 func DataSourceTMCCluster() *schema.Resource {
 	return &schema.Resource{
-		ReadContext: dataSourceTMCClusterRead,
-		Schema:      clusterSchema,
+		ReadContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+			return dataSourceClusterRead(context.WithValue(ctx, contextMethodKey{}, "read"), d, m)
```

Comment:
lets be more explicit here and call it dataSrcRead

Reply:
Done